File: Java-Core/src/main/java/presto/parser/OParserFactory.java
package presto.parser;

import java.io.IOException;
import java.io.InputStream;

import org.antlr.v4.runtime.ANTLRInputStream;



public class OParserFactory implements IParserFactory {

	@Override
	public ILexer newLexer(String data) {
		return new ONamingLexer(new ANTLRInputStream(data));
	}

	@Override
	public IParser newParser(String data) {
		return new OCleverParser(data);
	}

	@Override
	public ILexer newLexer(InputStream data) {
		try {
			return new ONamingLexer(new ANTLRInputStream(data));
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
	}

	@Override
	public IParser newParser(String path, InputStream data) {
		try {
			return new OCleverParser(path, data);
		} catch (IOException e) {
			throw new RuntimeException(e);
		}
	}

}

File: Java-CodeServer/src/main/resources/js/prompto/parser/OParserFactory.js
var antlr4 = require("antlr4/index");
var ONamingLexer = require("./ONamingLexer").ONamingLexer;
var OCleverParser = require("./OCleverParser").OCleverParser;

exports.OopsParserFactory =  function() {
	
	this.newLexer = function(data) {
		return new ONamingLexer(new antlr4.InputStream(data));
	};

	this.newParser = function(path, data) {
		return new OCleverParser(path, data);
	};

};